### PR TITLE
[KOGITO-6812] - Kogito operator should support label on deployment

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Many thanks for submitting your Pull Request :heart:! 

Jira issue : https://issues.redhat.com/browse/KOGITO-6812

with this improvement user can mark any kogito deployment object with `kogito.kie.org/runtime: 'true'` annotation and kogito operator will consider that deployment as `KogitoRuntime`. Kogito Operator will inject supporting service url into these deployment.

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [ ] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [ ] Pull Request contains a link to the JIRA issue
- [ ] Pull Request contains a description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've ran `make before-pr` and everything is working accordingly
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run operator BDD testing</b>
  Please add comment: <b>/jenkins test</b>

* <b>Run RHPAM operator BDD testing</b>
  Please add comment: <b>/jenkins RHPAM test</b>
</details>

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Comment by radtriste** [comment]
gonna try to give it a go today

File: bundle/app/manifests/kogito-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -739,6 +739,21 @@ spec:
           - patch
           - update
           - watch
+        - apiGroups:
```
Comment (sutaakar) [inline bundle/app/manifests/kogito-operator.clusterserviceversion.yaml:742]: Looks like a copy paste from entry above

File: config/rbac/app/role.yaml [hunk 2]
```diff
@@ -181,6 +181,21 @@ rules:
   - patch
   - update
   - watch
+- apiGroups:
```
Comment (sutaakar) [inline config/rbac/app/role.yaml:184]: Same here

**Comment by sutaakar** [comment]
Looks ok to me, will leave it on Tristan to test

**Comment by radtriste** [comment]
Tested ok. Config Maps are correctly bound to the deployment when annotation is there,

Just one remark: we are losing information that the operator has added the supporting services configmaps to the deployment.
@vaibhavjainwiz Would there be a way to add to the deployment status, letting the user know that the supporting services have been bound to the deployment ?

**Comment by radtriste** [comment]
also @vaibhavjainwiz could you add an example of a deployment with the annotation ?

This is the one I used for the tests. Feel free to take it and update it if needed
```yaml
kind: Deployment
apiVersion: apps/v1
metadata:
  name: test-deployment
  annotations:
    kogito.kie.org/runtime: 'true'
spec:
  replicas: 1
  selector:
    matchLabels:
      app: process-quarkus-example
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: process-quarkus-example
    spec:
      containers:
        - name: process-quarkus-example
          image: 'quay.io/kiegroup/examples-process-quarkus-example:latest'
          imagePullPolicy: Always
          ports:
            - containerPort: 8080
              protocol: TCP
          readinessProbe:
            httpGet:
              path: /q/health/ready
              port: 8080
              scheme: HTTP
            timeoutSeconds: 1
            periodSeconds: 10
            successThreshold: 1
            failureThreshold: 3
          livenessProbe:
            httpGet:
              path: /q/health/live
              port: 8080
              scheme: HTTP
            timeoutSeconds: 1
            periodSeconds: 10
            successThreshold: 1
            failureThreshold: 3
          startupProbe:
            httpGet:
              path: /q/health/live
              port: 8080
              scheme: HTTP
            timeoutSeconds: 1
            periodSeconds: 10
            successThreshold: 1
            failureThreshold: 3
          env:
          - name: RUNTIME_TYPE
            value: quarkus
```

**Comment by vaibhavjainwiz** [comment]
@radtriste I tried to add extra `DeploymentCondition` inside status of `Deployment` object to represent supporting service endpoint injection info but it give me error `the object has been modified;` I had tried all mean but not able to overcome this issue.

For user info, we could add a note in Kogito docs to check inserted configmap inside Deployment object.

For update example I had a Jira in queue https://issues.redhat.com/browse/KOGITO-6297. I will update it.

**Comment by radtriste** [comment]
> @radtriste I tried to add extra `DeploymentCondition` inside status of `Deployment` object to represent supporting service endpoint injection info but it give me error `the object has been modified;` I had tried all mean but not able to overcome this issue.
> 
> For user info, we could add a note in Kogito docs to check inserted configmap inside Deployment object.
> 
> For update example I had a Jira in queue https://issues.redhat.com/browse/KOGITO-6297. I will update it.

I guess this will be for later improvement then, if possible. Let's merge that one.

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
Can you address @sutaakar comment ?

File: bundle/app/manifests/kogito-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (vaibhavjainwiz) [reply bundle/app/manifests/kogito-operator.clusterserviceversion.yaml]: removed duplicate entries

File: config/rbac/app/role.yaml (hunk 2 above)
Comment (vaibhavjainwiz) [reply config/rbac/app/role.yaml]: removed duplicate entries

**Comment by vaibhavjainwiz** [comment]
> Can you address @sutaakar comment ?

@radtriste I had incorporated @sutaakar comments.

**Review by sutaakar — APPROVED** [review]

**Pull request merged**